sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -648,8 +648,24 @@ static int sof_audio_probe(struct platform_device *pdev)
 					  &sof_audio_card_rt5682);
 }
 
+static int sof_rt5682_remove(struct platform_device *pdev)
+{
+	struct snd_soc_card *card = platform_get_drvdata(pdev);
+	struct snd_soc_component *component = NULL;
+
+	for_each_card_components(card, component) {
+		if (!strcmp(component->name, rt5682_component[0].name)) {
+			snd_soc_component_set_jack(component, NULL, NULL);
+			break;
+		}
+	}
+
+	return 0;
+}
```

Comment:
@juimonen  a couple of comments: 1. a typo in the commit message "...which disables ..." instead of disabled. 2. Do you think cancelling the jack_detect_work in rt5682 codec driver's remove will be a better option? https://github.com/thesofproject/linux/blob/adbdb595d044f5b1ddcf6c81d620f9cb57ea2014/sound/soc/codecs/rt5682.c#L2339

Reply:
@ranjani ok, I'll fix the commit message. Adding the remove to sof_rt5682.c was suggested by @bardliao so I'm not totally sure of the possible reason... I guess the removal starts from bottom up and this is the way to do it in earlier stage? however I can try to add it solely to codec and see what happens...

Reply:
@juimonen I "guess" the reason is we reset the codec on rt5682_remove(), which reset the codec IRQ pin to the initial gpio function and trigger the interrupt. And we doesn't disable irq before that.